File: backend/src/review/objects/MerchantReview.scala
package delivery.review.objects

import delivery.shared.objects.{MerchantId, OrderId, UserId}

final case class MerchantReview(
    id: String,
    orderId: OrderId,
    merchantId: MerchantId,
    customerId: UserId,
    customerName: String,
    rating: Int,
    description: String,
    imageUrl: Option[String],
    upvotes: Int,
    downvotes: Int,
    createdAt: String,
    orderItemNames: List[String] = Nil
)

File: frontend/src/objects/review/MerchantReview.ts
import type { MerchantId, OrderId, UserId } from '@/objects/shared/ids'

export interface MerchantReview {
  id: string
  orderId: OrderId
  merchantId: MerchantId
  customerId: UserId
  customerName: string
  rating: number
  description: string
  imageUrl?: string | null
  upvotes: number
  downvotes: number
  createdAt: string
  orderItemNames?: string[]
}
